op-service/sources/eth_client.go

```diff
@@ -170,22 +123,32 @@ func NewEthClient(client client.RPC, log log.Logger, metrics caching.Metrics, co
 	if err := config.Check(); err != nil {
 		return nil, fmt.Errorf("bad config, cannot create L1 source: %w", err)
 	}
+	recCfg := RPCReceiptsConfig{
+		MaxBatchSize:        config.MaxRequestsPerBatch,
+		ProviderKind:        config.RPCProviderKind,
+		MethodResetDuration: config.MethodResetDuration,
+	}
+
 	client = LimitRPC(client, config.MaxConcurrentRequests)
+	recProvider := NewCachingRPCReceiptsProvider(client, log, recCfg, metrics, config.ReceiptsCacheSize)
```

Comment:
Learning about our config, does this mean node operators will have a few more fields to add? What's the behavior if they don't?

Reply:
In its current form, the PR doesn't change the configuration of the `EthClient`. The `EthClientConfig` struct is untouched and is passed unchanged to `NewEthClient`.